accepted/2025/PrunePackageReference-with-direct-PackageReference.md

```diff
@@ -0,0 +1,216 @@
+
+Given that pruning is an offline operation, ie NuGet does not talk to the sources for pruned packages, we never validate that the requested package exists and the specified version may be incorrect. This is likely to not be of too much consequence, since it's only really possible with hand-editing, and the impact of it is minimal since the package is not used anyways.
+
+### Should we prune direct PackageReference
+
+When the NU1510 warning is raised, we are giving the customer a clear signal that they need to remove the PackageReference.
+In these scenarios, whether we prune direct package references is not very critical.
+
+In the partial prune scenarios such as 2b and 2c, pruning could be beneficial in the .NET leg, helping avoid additional audit warnings.
+However, due to the fact that the PackageReference is in other frameworks, the benefit is probably more minimal, since if the package itself had a vulnerability, the audit warning will be warranted since the customer is using it in a case where it is not being pruned.
+There are a few things that pruning of direct packages introduces:
+
+- The new concept of a pruned direct package. For transitive pruning, the end user would not see anything beyond a reduction in packages downloaded. For direct packages, the PM UI, list & why commands.
+
+The biggest drawback is cost. Direct pruning requires a lot of parts to be updated and the benefit is minimal. Only 5% of .NET SDK based projects are suspected to be multi-targeted and data analysis would suggest only 1-2% of projects could ever fall into this scenario and all of that disregards the per framework considerations.
+
+## Drawbacks
+
+- Additional UX concepts for both 1st and 3rd party tooling that manage packages.
+- Cost of direct PackageReference pruning vs the potential benefit since customers in affected scenarios with a vulnerable package would either get a NU1510 or get a legit audit warning.
+
+## Rationale and alternatives
+
+- Update the NU1510 heuristic only. Retain the current behavior for pruning.
+- Instead of removing the direct PackageReference from the graph completely, treat it as `IncludeAssets="none"`. This would allow get us benefits from Component Governance, in a more cost-effective way. `NuGetAudit` would also need to be updated to stop auditing unused packages, similarly to how Component Governance approaches the problem. [Make ExcludeAssets visible in Audit and nuget why](https://github.com/NuGet/Home/issues/13860)
+
+## Prior Art
+
+N/A
+
+## Unresolved Questions
+
+- Should we prune direct packages? If the answer is yes, the following unresolved questions apply:
+  - What should `dotnet list package` show for direct pruned packages? Should it show the version specified as requested?
```

Comment:
It could list a symbol, or something like `#1`, and under the table say that it means pruned (and an aka.ms link to learn what pruning is?)

Reply:
It'd have to be something similar to (A) that we get for auto referenced.

I'll explicitly add a view of how that looks like today.